core/src/main/scala/zio/config/ConfigDocsModule.scala

```diff
@@ -420,6 +420,12 @@ trait ConfigDocsModule extends WriteModule {
    * from various sources.
    */
   final def generateDocs[A](config: ConfigDescriptor[A]): ConfigDocs = {
+    def lookAhead(desc: LazyConfigDescriptor[_]): ConfigDescriptor[_] =
```

Comment:
I see where this is going. Nice one on understanding this verbose logic and fixing the issue. Thanks @vigoo 

FYI (and probably unrelated): 

I am making some very minor changes such that `LazyConfigDescriptor ` becomes one of the terms in the ADT, but rest of the code that makes recursion work remains the same. 

https://github.com/zio/zio-config/blob/api_docs/core/src/main/scala/zio/config/ConfigDescriptorModule.scala#L1317